projects/libs/core-library-typescript/test/consensus/block-builder.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { types } from "../../src/common-library/types";
+import * as chai from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import * as sinonChai from "sinon-chai";
+import { stubInterface } from "ts-sinon";
+import BlockBuilder from "../../src/consensus/block-builder";
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+const transactionPool = stubInterface<types.TransactionPoolClient>();
+const virtualMachine = stubInterface<types.VirtualMachineClient>();
+
+function randomUInt() {
+  return Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+}
+
+function aRandomTransaction(): types.Transaction {
+    return {
+        version: 1,
+        sender: randomUInt().toString(16),
+        contractAddress:  randomUInt().toString(16),
+        payload: Math.random().toString(),
+        signature: ""
+    };
+}
+
+function aDummyTransactionSet(numberOfTransactions = 3) {
+  const transactions = [];
+  for (let i = 0; i < numberOfTransactions; i++); {
+    transactions.push(aRandomTransaction);
+  }
+
+  return transactions;
+}
+
+function aRandomStateDiff(): types.modifiedStateKey[] {
+  return [
+    {
+      contractAddress: randomUInt().toString(16),
+      key: `key-${randomUInt()}`,
+      value: randomUInt().toString(),
+    }
+  ];
+}
+
+const dummyTransactionSet = aDummyTransactionSet();
+
+const pullAllPendingTransactionsOutput: types.GetAllPendingTransactionsOutput = { transactions: dummyTransactionSet };
+transactionPool.getAllPendingTransactions.returns(pullAllPendingTransactionsOutput);
+
+const processTransactionSetOutput: types.ProcessTransactionSetOutput = {
+  processedTransactions: dummyTransactionSet,
+  stateDiff: aRandomStateDiff()
+};
+virtualMachine.processTransactionSet.returns(processTransactionSetOutput);
+
+const blockBuilder = new BlockBuilder({ virtualMachine, transactionPool });
+
+describe("a block", () => {
+  it("is built with the transactions on transaction pool", async () => {
+    const block = await blockBuilder.buildNextBlock(1);
+
+    block.transactions.should.eql(dummyTransactionSet);
```

Comment:
I've started this discussion with @netoneko as well: we should use the mode modern `expect` instead of `should`, since the former supports many additional capabilities and has a larger community of tools and contributors. In this case, it won't matter that much, but if we won't choose one - we'd find ourselves duplicating testing tools and primitives all over.